### PR TITLE
[Model] fix GCN

### PR DESCRIPTION
**Pull request by ZiyueHuang**
## Description
Add mxnet GCN spmv, as README mentioned. @zheng-da @eric-haibin-lin 

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [Model], [Doc], [Feature]])
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented
- [x] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [x] Related issue is referred in this PR

### CODE REVIEW

**Comment by ZiyueHuang** [comment]
# Final accuracy 87.17% (same as gcn_batch.py)
DGLBACKEND=mxnet python gcn_spmv.py --dataset "citeseer" --n-epochs 200 --gpu 1 --n-layers 10 --normalization 'sym' --self-loop

File: examples/mxnet/gcn/README.md [hunk 1]
```diff
@@ -10,6 +10,11 @@ The folder contains two implementations of GCN. `gcn_batch.py` uses user-defined
 message and reduce functions. `gcn_spmv.py` uses DGL's builtin functions so
 SPMV optimization could be applied.
 
+The provided implementation is a bit different from the original paper for better performance.
+For example, the current node vector and the accumulated messages are concatenated together
+as the input for next GCN layer
+([line](https://github.com/dmlc/dgl/blob/master/examples/mxnet/gcn/gcn_batch.py#L46)).
```
Comment (zheng-da) [inline examples/mxnet/gcn/README.md]: i'm not sure if the line will change afterwards.

File: examples/mxnet/gcn/gcn_spmv.py [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+                              for i in range(len(data.graph))])
+    out_degree = mx.nd.array([data.graph.out_degree(i)
+                              for i in range(len(data.graph))])
+
+    in_feats = features.shape[1]
+    n_classes = data.num_labels
+    n_edges = data.graph.number_of_edges()
+
+    if args.gpu <= 0:
+        cuda = False
```
Comment (zheng-da) [inline examples/mxnet/gcn/gcn_spmv.py]: please remove this. it's my fault. When i ported the GCN impl from Pytorch, I forgot to remove this line.

File: examples/mxnet/gcn/gcn_spmv.py [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+        ctx = mx.cpu(0)
+    else:
+        cuda = True
```
Comment (zheng-da) [inline examples/mxnet/gcn/gcn_spmv.py]: the same here.

File: examples/mxnet/gcn/gcn_spmv.py [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+        trainer.step(features.shape[0])
+
+        if epoch >= 3:
+            dur.append(time.time() - t0)
+            print("Epoch {:05d} | Loss {:.4f} | Time(s) {:.4f} | ETputs(KTEPS) {:.2f}".format(
+                epoch, loss.asnumpy()[0], np.mean(dur), n_edges / np.mean(dur) / 1000))
+
+    # test set accuracy
+    pred = model(features)
+    accuracy = (pred*100).softmax().pick(labels).mean()
```
Comment (zheng-da) [inline examples/mxnet/gcn/gcn_spmv.py]: this accuracy includes the training data. could you please avoid using the training data for test?

**Comment by ylfdq1118** [comment]
@jermainewang actually implemented gcn_spmv for mxnet. One thing very important that minjie did was make sure the code is an exact match of author's implementation. Maybe @jermainewang you incorporate your changes in this PR?

**Comment by ZiyueHuang** [comment]
It’s fine. I can fix gcn_batch.py, which has not been done in minjie’s PR.
